hikari/api/rest.py

```diff
@@ -2011,23 +2022,105 @@ def fetch_audit_log(
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            The guild to fetch the audit logs from. This can be a
+            guild object or the ID of an existing guild.
+        before : hikari.undefined.UndefinedOr[hikari.snowflakes.SearchableSnowflakeishOr[hikari.snowflakes.Unique]]
+            If provided, filter to only actions after this one.
+        user : hikari.undefined.UndefinedOr[hikari.snowflakes.SnowflakeishOr[hikari.users.PartialUser]]
+            If provided, the user to filter for.
+        event_type : hikari.undefined.UndefinedOr[hikari.audit_logs.AuditLogEventType]
+            If provided, the event type to filter for.
+
+        Returns
+        -------
+        hikari.iterators.LazyIterator[hikari.audit_logs.AuditLog]
+            The guild's audit log.
+
+        !!! note
+           This call is not a coroutine function, it returns a special type of
+           lazy iterator that will perform API calls as you iterate across it.
+           See `hikari.iterators` for the full API for this iterator type.
+
+        Raises
+        ------
+        hikari.errors.BadRequestError
+            If any of the fields that are passed have an invalid value.
+        hikari.errors.ForbiddenError
+            If you are missing the `VIEW_AUDIT_LOG` permission.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+
+        !!! note
+            The exceptions on this endpoint will only be raised once the
+            result is awaited or interacted with. Invoking this function
+            itself will not raise anything.
+        """
 
     @abc.abstractmethod
     async def fetch_emoji(
         self,
         guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
-        # This is an emoji ID, which is the URL-safe emoji name, not the snowflake alone.
-        # likewise this only is valid for custom emojis, unicode emojis make little sense here.
-        emoji: typing.Union[str, emojis.CustomEmoji],
+        emoji: snowflakes.SnowflakeishOr[emojis.CustomEmoji],
     ) -> emojis.KnownCustomEmoji:
-        ...
+        """Fetch a guild emoji.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            The guild to fetch the emoji from. This can be a
+            guild object or the ID of an existing guild.
+        emoji : hikari.snowflakes.SnowflakeishOr[hikari.emojis.CustomEmoji]
+            The emoji to fetch. This can be a `hikari.emojis.CustomEmoji`
+            or the ID of an existing emoji.
+
+        Returns
+        -------
+        hikari.emojis.KnownCustomEmoji
+            The requested emoji.
+
+        Raises
+        ------
+        hikari.errors.NotFoundError
+            If the guild or the emoji are not found.
```

Comment:
or if you aren't in the guild maybe? not sure if the nuanced difference between a guild not being found vs a guild not being found in the group of guilds you have access to would be worth explicitly documenting both cases just felt like pointing that out.

Reply:
Sometimes there are, sometimes they aren't. I'm pretty sure I documented them all